server/functions/src/index.ts

```diff
@@ -20,10 +20,229 @@ export const createUserInDatabase = functions.auth
+      console.error("Team data not found!");
+      return undefined;
+    }
+
+    const team1Info = team1Data.data();
+    const team2Info = team2Data.data();
+
+    if (!team1Info || !team2Info) {
+      console.error("Team info is undefined!");
+      return undefined;
+    }
+
+    const startDate = await getNextSundayDate();
+    const matchupRef = admin.firestore().collection("matchups").doc();
+
+    const matchupData: any = {
+      id: matchupRef.id,
+      createdAt: new Date(),
+      status: "active",
+      weekStartDate: startDate,
+      homeUserId: userId1,
+      awayUserId: userId2,
+    };
+
+    const start = new Date(startDate);
+    for (let i = 0; i < 7; i++) {
+      const date = new Date(start.getTime() + i * 24 * 60 * 60 * 1000);
+      const dateStr = date.toISOString().split("T")[0];
+
+      matchupData[dateStr] = {
+        homeTeam: {
+          id: teamId1,
+          score: 0,
+          lineup: team1Info.lineup || [],
+          qualifyingPlayers: team1Info.lineup || [],
+          metadata: {},
+        },
+        awayTeam: {
+          id: teamId2,
+          score: 0,
+          lineup: team2Info.lineup || [],
+          qualifyingPlayers: team2Info.lineup || [],
+          metadata: {},
+        },
+      };
+    }
+
+    await matchupRef.set(matchupData);
+    return matchupRef.id;
+  } catch (error) {
+    console.error("Error creating weekly matchup:", error);
+    return undefined;
+  }
+};
+
+const getNextSundayDate = async (): Promise<string> => {
+  const today = new Date();
+  const dayOfTheWeek = today.getDay();
+
+  if (dayOfTheWeek === 0) {
+    const hasGamesStarted = await checkIfSundayGamesStarted();
+
+    if (!hasGamesStarted) {
+      return today.toISOString().split("T")[0];
+    } else {
+      const nextSunday = new Date(today.getTime() + 7 * 24 * 60 * 60 * 1000);
+      return nextSunday.toISOString().split("T")[0];
+    }
+  }
+
+  const daysUntilSunday = 7 - dayOfTheWeek;
+
+  const nextSunday = new Date(
+    today.getTime() + daysUntilSunday * 24 * 60 * 60 * 1000,
+  );
+  return nextSunday.toISOString().split("T")[0];
+};
+
+const checkIfSundayGamesStarted = async (): Promise<boolean> => {
+  try {
+    const today = new Date();
+    const todayStr = today.toISOString().split("T")[0];
+
+    const response = await fetch(
+      `${process.env.BALLDONTLIE_GAMES_URL}${todayStr}`,
+      {
+        headers: {
+          Authorization: `${process.env.BALLDONTLIE_API_KEY}`,
+        },
+      },
+    );
+
+    if (!response.ok) {
+      console.log("Failed fetch for NBA games");
+      return true; // Default to true (games started) to use next Sunday
+    }
+
+    const data: any = await response.json();
+    const games = data.data || [];
+
+    if (games.length === 0) {
+      return false;
+    }
+
+    const gameTimes = games.map((game: any) => new Date(game.status));
+    const earliestGame = new Date(
+      Math.min(...gameTimes.map((d: any) => d.getTime())),
+    );
```

Comment:
This is more of a small nitpick but you can improve this
```suggestion
    const earliestGame = new Date(
      Math.min(...games.map(g => new Date(g.status).getTime()))
);
```

Reply:
Bet, will do with this suggestion